### ISSUE TITLE
GHC panic if SMT solver is missing

### ISSUE THREAD

**Issue by Gabriel439** [ucsd-progsys/liquidhaskell#636]
For some strange reason, if there is no SMT solver on the path `liquid` fails with a GHC panic.  I got a full stack trace of the problem:

```haskell
*** Exception (reporting due to +RTS -xc): (base:GHC.Exception.SomeException), stack trace: 
  Language.Haskell.Liquid.Types.Errors.panicDoc,
  called from Language.Haskell.Liquid.Types.Errors.panic,
  called from Language.Haskell.Liquid.UX.CmdLine.withSmtSolver,
  called from Language.Haskell.Liquid.UX.CmdLine.getOpts,
  called from DynFlags.CAF
liquid: liquid: panic! (the 'impossible' happened)
  (GHC version 7.10.3 for x86_64-apple-darwin):
	v_unsafeGlobalDynFlags: not initialised
Stack trace:
  DynFlags.CAF (<entire-module>)

Please report this as a GHC bug:  http://www.haskell.org/ghc/reportabug
```

One of my colleagues at work ran into this error and the problem went away when they installed `z3`.

My rough understanding of how the problem arises is this:

* The user has no solver on their path
* Liquid Haskell tries to throw an exception saying that the user has no solver on their path
* In the process of throwing that exception it tries to access some information from the `ghc` API
* The use of `ghc`'s API is incorrect and (indirectly) causes `ghc` to access the `v_unsafeGlobalDynFlags` "global variable" before it is set, and it's programmed to `panic` in that case

There are more details about when it's safe to access that information here:

* https://github.com/ghc/ghc/blob/2f45cf3f48162a5f843005755dafa1c5c1b451a7/compiler/main/DynFlags.hs#L4829
* https://github.com/ghc/ghc/blob/37945c1db2f893657c1e3b9b26704cbf3ef27a5a/compiler/main/CmdLineParser.hs#L301

**Comment by ranjitjhala** [comment]
As it happens we noticed this recently and @spinda has just fixed this in develop?

**Comment by spinda** [comment]
Yes, this was fixed in #634. I believe there will be a new release to Hackage sometime soon.

**Comment by Gabriel439** [comment]
Yay!  Thank you :smile:
